src/gui/screen_menu_odometer.cpp

```diff
@@ -0,0 +1,61 @@
+// screen_menu .cpp
+
+#include <stdlib.h>
+
+#include "screen_menus.hpp"
+#include "screen_menu.hpp"
+#include "i18n.h"
+#include "odometer.hpp"
+
+static const constexpr HelperConfig HelpCfg = { 10, IDR_FNT_NORMAL };
+enum : int {
+    TEXT_MAX_LENGTH = 150
+};
+
+using Screen = ScreenMenu<EHeader::On, EFooter::On, HelpCfg, MI_RETURN>;
+
+class ScreenMenuOdometer : public Screen {
+    char text[TEXT_MAX_LENGTH];
+
+public:
+    static const constexpr char *label = N_("ODOMETER");
+    ScreenMenuOdometer();
+};
+
+int first_decimal(float f) {
+    return (int)(f * 10) % 10;
+}
+
+ScreenMenuOdometer::ScreenMenuOdometer()
+    : Screen(_(label)) {
+
+    header.SetIcon(IDR_PNG_info_16px);
+    odometer.force_to_eeprom();
+    const float x = odometer.get(0) * .001f;
+    const float y = odometer.get(1) * .001f;
+    const float z = odometer.get(2) * .001f;
+    const float e = odometer.get(3) * .001f;
+
+    /// FIXME this is not aligned if "Filament" is translated or numbers have different length
```

Comment:
you can use printf("%12.5s... or even -%12.5s to align text inside the printf facility. Look at the documentation of formatting possibilities